packages/back-end/src/util/legacyAudit/base.ts

```diff
@@ -0,0 +1,93 @@
+import { UnionToIntersection } from "../types";
+
+import {
+  entityEvents as auditEvents,
+  EntityType as AuditEventResource,
+  EntityEventTemplate as AuditEventNameTemplate,
+} from "../../types/Audit";
+
+export {
+  entityEvents as auditEvents,
+  EntityType as auditEventResources,
+  EntityType as AuditEventResource,
+  EntityEventTemplate as AuditEventNameTemplate,
+} from "../../types/Audit";
+
+export type AuditNotificationEventMap<
+  Resource extends AuditEventResource,
+  Event extends AuditEventNameTemplate<Resource>
+> = Resource extends "savedGroup"
+  ? Event
+  : Resource extends "archetype"
+  ? Event
+  : Event extends "create"
+  ? "created"
+  : Event extends "update"
+  ? "updated"
+  : Event extends "delete"
+  ? "deleted"
+  : Event;
+
+export const auditNotificationEvent = <
```

Comment:
The type of this function is forced. It should be proof-read to make sure it actually implements what it says it does.